server.js

```diff
@@ -50,10 +53,37 @@ app.get('/block-me/server-sent-events', (req, res) => {
 
 // dummy CSP report endopoint
 app.post('/block-me/csp', (req, res) => {
+    res.set('Timing-Allow-Origin', '*');
     return res.sendStatus(200);
 });
 
 // reflects request headers back
 app.get('/reflect-headers', (req, res) => {
+    res.set('Access-Control-Allow-Origin', req.headers.origin || '*');
+    res.set('Access-Control-Allow-Credentials', 'true');
+    res.set('Timing-Allow-Origin', '*');
+
     return res.json({headers: req.headers});
+});
+
+// sets a cookie with provided value
+app.get('/set-cookie', (req, res) => {
+    res.set('Access-Control-Allow-Origin', req.headers.origin || '*');
+    res.set('Access-Control-Allow-Credentials', 'true');
```

Comment:
we need to set both of those of that for the cookie to be actually set nowadays. And we can't use '*' for ACAO. TIL!